File: CompilationTest/FakeEnumTest/results/FakeEnumTest.go
package main

import "fmt"

type TokenType struct {
}

func NewTokenType() *TokenType {
    this := new(TokenType)
    return this
}

var TokenTypeEndToken string = "EndToken";
var TokenTypeWhitespace string = "Whitespace";
var TokenTypeIdentifier string = "Identifier";
var TokenTypeOperatorX string = "Operator";
var TokenTypeNoInitializer string;

func main() {
    casingTest := TokenTypeEndToken
    fmt.Println(casingTest)
}

File: CompilationTest/FakeEnumTest/results/FakeEnumTest.pl
use strict;
use warnings;

package TokenType;

sub new
{
    my $class = shift;
    my $self = {};
    bless $self, $class;
    return $self;
}

our $end_token = "EndToken";
our $whitespace = "Whitespace";
our $identifier = "Identifier";
our $operator_x = "Operator";
our $no_initializer;

my $casing_test = $TokenType::end_token;
print(($casing_test) . "\n");